src/components/Table/Table.tsx

```diff
@@ -81,7 +81,10 @@ function Table<T>({
 
   const [sorting, setSorting] = useState<Sort>(initialSort);
 
-  const columnCount = onCheck ? columns.length + 1 : columns.length;
+  let columnCount = onCheck ? columns.length + 1 : columns.length;
+
+  const hasExpandableRows = data.some(row => !!row.expanded);
+  columnCount = hasExpandableRows ? columnCount + 1 : columnCount;
```

Comment:
Better to do something like this instead of mutating
```
const getColumnCount = () => {
    if(!onCheck && !hasExpandableRows){
        return columns.length;
    } else if (onCheck && hasExpandableRows) {
        return columns.length + 2;
    } else {
        return columns.length + 1;
    }
}
```

Reply:
i agree and shall i suggest this as it defines what is boolean by name and type and has no dependency in other scopes

```
const getColumnCount = (columns, onCheck, hasExpandableRows) => {
    if(!Boolean(onCheck) && !hasExpandableRows){
        return columns.length;
    }
    if (Boolean(onCheck) && hasExpandableRows) {
        return columns.length + 2;
    }
    
    return columns.length + 1;
}
```